packages/build/tests/plugins/snapshots/tests.js.md

```diff
@@ -2318,6 +2318,130 @@ Generated by [AVA](https://ava.li).
+    ────────────────────────────────────────────────────────────────␊
+    ␊
+    > Version␊
+      @netlify/build 1.0.0␊
+    ␊
+    > Flags␊
+      debug: true␊
+      featureFlags: {}␊
+      repositoryRoot: /file/path␊
+      testOpts:␊
+        pluginsListUrl: /file/path␊
+        silentLingeringProcesses: true␊
+    ␊
+    > Current directory␊
+      /file/path␊
+    ␊
+    > Config file␊
+      /file/path␊
+    ␊
+    > Resolved config␊
+      build:␊
+        environment:␊
+          - TEST␊
+          - PRODUCTION_TEST␊
+      plugins:␊
+        - inputs: {}␊
+          origin: config␊
+          package: netlify-plugin-contextual-env␊
+    ␊
+    > Context␊
+      production␊
+    ␊
+    > Available plugins␊
+    ␊
+    > Installing plugins␊
+       - netlify-plugin-contextual-env@1.0.0␊
+    ␊
+    > Loading plugins␊
+       - netlify-plugin-contextual-env@1.0.0 from netlify.toml␊
+    ␊
+    ────────────────────────────────────────────────────────────────␊
+      1. onPreBuild command from netlify-plugin-contextual-env      ␊
+    ────────────────────────────────────────────────────────────────␊
+    ␊
+    Exporting TEST=test.␊
+    Replaced 1 ENVs␊
+    ␊
+    (netlify-plugin-contextual-env onPreBuild completed in 1ms)␊
+    ␊
+    ────────────────────────────────────────────────────────────────␊
+      Netlify Build Complete                                        ␊
+    ────────────────────────────────────────────────────────────────␊
+    ␊
+    (Netlify Build completed in 1ms)`
+
+## Plugins can specify non-matching compatibility.nodeVersion
+
+> Snapshot 1
+
+    `␊
+    ────────────────────────────────────────────────────────────────␊
+      Netlify Build                                                 ␊
+    ────────────────────────────────────────────────────────────────␊
+    ␊
+    > Version␊
+      @netlify/build 1.0.0␊
+    ␊
+    > Flags␊
+      debug: true␊
+      featureFlags: {}␊
+      repositoryRoot: /file/path␊
+      testOpts:␊
+        pluginsListUrl: /file/path␊
+        silentLingeringProcesses: true␊
+    ␊
+    > Current directory␊
+      /file/path␊
+    ␊
+    > Config file␊
+      /file/path␊
+    ␊
+    > Resolved config␊
+      build:␊
+        environment:␊
+          - TEST␊
+          - PRODUCTION_TEST␊
+      plugins:␊
+        - inputs: {}␊
+          origin: config␊
+          package: netlify-plugin-contextual-env␊
+    ␊
+    > Context␊
+      production␊
+    ␊
+    > Available plugins␊
+    ␊
+    > Installing plugins␊
+       - netlify-plugin-contextual-env@1.0.0␊
+    ␊
+    > Loading plugins␊
+       - netlify-plugin-contextual-env@1.0.0 from netlify.toml␊
+    ␊
+    ────────────────────────────────────────────────────────────────␊
+      1. onPreBuild command from netlify-plugin-contextual-env      ␊
+    ────────────────────────────────────────────────────────────────␊
+    ␊
+    Exporting TEST=test.␊
+    Replaced 1 ENVs and wrote .env file␊
```

Comment:
This is the difference in the test snapshot (since we are normalizing versions :-/). The `0.2.0` and `0.3.0` versions of that plugin have different outputs there. This test is a little hacky :/

Reply:
True :/

Maybe it would be worth looking into the `version` normalization and see if it can be made optional or configurable on our side? Because I think in the long run we'll probably start exercising more features around versioning and it hasn't been the first time we're seeing this. (On a separate PR/Issue ofc 😄)

Reply:
Yes, this was my feeling as well. We allow opting out of it. :+1: